localstack-core/localstack/utils/xray/trace_header.py

```diff
@@ -0,0 +1,192 @@
+# This file is part of LocalStack.
+# It is adapted from aws-xray-sdk-python licensed under the Apache License 2.0.
+# You may obtain a copy of the Apache License 2.0 at http://www.apache.org/licenses/LICENSE-2.0
+# Original source: https://github.com/aws/aws-xray-sdk-python/blob/master/aws_xray_sdk/core/models/trace_header.py
+# Modifications:
+# * Add optional lineage field for https://docs.aws.amazon.com/lambda/latest/dg/invocation-recursion.html
+# * Add ensure_root_exists(), ensure_parent_exists(), and ensure_sampled_exists()
+# * Add generate_random_id() from https://github.com/aws/aws-xray-sdk-python/blob/d3a202719e659968fe6dcc04fe14c7f3045b53e8/aws_xray_sdk/core/models/entity.py#L308
+
+import binascii
+import logging
+import os
+
+from localstack.utils.xray.traceid import TraceId
+
+log = logging.getLogger(__name__)
+
+ROOT = "Root"
+PARENT = "Parent"
+SAMPLE = "Sampled"
+SELF = "Self"
+LINEAGE = "Lineage"
+
+HEADER_DELIMITER = ";"
+
+
+def generate_random_id():
+    """
+    Generate a random 16-digit hex str.
+    This is used for generating segment/subsegment id.
+    """
+    return binascii.b2a_hex(os.urandom(8)).decode("utf-8")
```

Comment:
I know this is taken from AWS code, but I think we could use some Python utils to make it a bit nicer.

`from secrets import token_hex`

```suggestion
    return token_hex(8)
```

Reply:
nice suggestions ✨ This looks much cleaner 🤩 

I fully agree on a technical level. For licensing and maintenance reasons, I opted to minimize changes to the original AWS code here (it is easier to match the authoritative AWS implementation regarding header parsing). Do you think it's worth deviating here?

Reply:
I suppose not, we could also keep the code as it is, if it makes it easier to port new changes as they come (not sure how often this happens?).

Reply:
I'm also crying every time I see `%s%s%s%s%s` 😭 . Nevertheless, I tend to prefer keeping the original for full parity (it's easier to relate to the AWS repo). Yeah, I don't expect direct changes here, maybe indirect ones for other X-Ray parts from the same source repo.